spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/MultiMethodRabbitListenerEndpoint.java

```diff
@@ -47,7 +52,36 @@ protected HandlerAdapter configureListenerAdapter(MessagingMessageListenerAdapte
 			invocableHandlerMethods.add(getMessageHandlerMethodFactory()
 					.createInvocableHandlerMethod(getBean(), method));
```

Comment:
Yes, that works, too. Since we shot it already, we can't remove ctor. So, yes, delegate to the setter is good compromise.